src/core_modules/capture-core/components/Pages/MainPage/TeiWorkingLists/epics/teiViewEpics/lib/initTeiWorkingListsView.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+import log from 'loglevel';
+import i18n from '@dhis2/d2-i18n';
+import { errorCreator } from 'capture-core-utils';
+import { getTeiListData } from './getTeiListData';
+import {
+    initListViewSuccess,
+    initListViewError,
+} from '../../../../WorkingListsCommon';
+import type { Input } from './initTeiWorkingListsView.types';
+
+const getSortByConfig = (columnsMetaForDataFetching) => {
+    let column = columnsMetaForDataFetching.find(col => col.visible);
+    if (!column) {
+        column = columnsMetaForDataFetching[0];
+    }
+
+    return {
+        sortById: column.id,
+        sortByDirection: 'desc',
+    };
+};
+
+export const initTeiWorkingListsView = ({
+    programId,
+    orgUnitId,
+    storeId,
+    columnsMetaForDataFetching,
+    querySingleResource,
+    absoluteApiPath,
+}: Input) => {
+    const { sortById, sortByDirection } = getSortByConfig([...columnsMetaForDataFetching.values()]);
```

Comment:
This can be read in two ways. Either "get sort by the provided configuration" or "get sort-by configuration". 

Now this is again very arguable and I understand this is a tricky name.
If I had to think of a name for this, I would go for `getConfigForSorting`

Please feel free to resolve this comment if this doesn't make sense to you.

Reply:
Fair point, changing to `getSortConfig`.